immich/docker-compose.yaml

```diff
@@ -0,0 +1,65 @@
+services:
+  immich:
+    image: ${IMMICH_IMAGE:-ghcr.io/immich-app/immich-server:release}
+    volumes:
+      - ${IMMICH_UPLOAD_LOCATION}:/usr/src/app/upload
```

Comment:
I don't really like this style of env var substitution, it puts too much required knowledge onto the editor of the .env file. 

Instead, I would create a config like "IMMICH_USE_EXTERNAL_VOLUME=true" and "IMMICH_EXTERNAL_VOLUME=/path/to/photos" 

In the template you can discriminate on IMMICH_USE_EXTERNAL_VOLUME and configure the volume this way:

```
version: '3.8'

services:
  my_service:
    image: my_image
    volumes:
      - photos:/app/data

volumes:
  photos:
    driver: local
    driver_opts:
      type: none
      o: bind
      device: /custom/path/on/host
```

that way the named volume is always named photos.

Furthermore, all the other volumes should also be named inside the `volumes` section.

Reply:
I re-did the functionality surrounding mapping a bind mount vs. a docker volume ([8f077bd](https://github.com/EnigmaCurry/d.rymcg.tech/pull/271/commits/8f077bdcafbdfd673a868dfcf5132701e91c36f3)).

Now, though, the container is recognizing neither the docker volume _nor_ the bind mount.

Reply:
My mistake: the container recognizes both docker volumes and bind mounts.